From 1ceffec2523888b4701405b7cf1710489e17011a Mon Sep 17 00:00:00 2001
From: Scott Mercer <TheRootEd24@gmail.com>
Date: Wed, 19 Mar 2025 06:31:44 -0400
Subject: [PATCH] ipq50xx: packages: dumpimage: fix compile patch

This patch is to fix compilation for uboot-tool V2025.01 in openwrt.

* same "compile for enviroment patch" to makefile, that is used
  in uboot-envtools.

* remove mkeficapsule from build to avoid gnulib dependencies

Signed-off-by: Scott Mercer <TheRootEd24@gmail.com>

Signed-off-by: lean <chenlin@fnnas.com>
---

--- a/tools/Makefile
+++ b/tools/Makefile
@@ -22,6 +22,26 @@
 #    That's as long as the features of those tools aren't modified.
 #
 
+override HOSTCC = $(CC)
+
+ifneq ($(TARGET_CFLAGS),)
+KBUILD_HOSTCFLAGS = $(TARGET_CFLAGS)
+endif
+ifneq ($(TARGET_LDFLAGS),)
+KBUILD_HOSTLDFLAGS = $(TARGET_LDFLAGS)
+endif
+
+# Compile for a hosted environment on the target
+HOST_EXTRACFLAGS  = -I$(srctree)/tools \
+		$(patsubst -I%,-idirafter%, $(filter -I%, $(UBOOTINCLUDE))) \
+		-idirafter $(srctree)/tools/env \
+		-DUSE_HOSTCC \
+		-DTEXT_BASE=$(TEXT_BASE)
+
+ifeq ($(MTD_VERSION),old)
+HOST_EXTRACFLAGS += -DMTD_OLD
+endif
+
 # Enable all the config-independent tools
 ifneq ($(HOST_TOOLS_ALL),)
 CONFIG_ARCH_KIRKWOOD = y
@@ -327,8 +347,7 @@
 		-I$(srctree)/scripts/dtc/libfdt \
 		-I$(srctree)/tools \
 		-DUSE_HOSTCC \
-		-D__KERNEL_STRICT_NAMES \
-		-D_GNU_SOURCE
+		-D__KERNEL_STRICT_NAMES
 
 ifeq ($(CROSS_BUILD_TOOLS),)
 __build:	$(LOGO-y)
